### PR TITLE
fix: Update `ptn/azd/aks` parameter

### PR DESCRIPTION
**Pull request by Menghua1**
## Description
Updated some aks parameters.

## Pipeline Reference

| Pipeline |
| -------- |
|    [avm.ptn.azd.aks](https://github.com/Menghua1/bicep-registry-modules/actions/workflows/avm.ptn.azd.aks.yml)      |

## Type of Change

- [ ] Update to CI Environment or utilities (Non-module affecting changes)
- [x] Azure Verified Module updates:
  - [ ] Bugfix containing backwards-compatible bug fixes, and I have NOT bumped the MAJOR or MINOR version in `version.json`:
    - [ ] Someone has opened a bug report issue, and I have included "Closes #{bug_report_issue_number}" in the PR description.
    - [ ] The bug was found by the module author, and no one has opened an issue to report it yet.
  - [ ] Feature update backwards compatible feature updates, and I have bumped the MINOR version in `version.json`.
  - [ ] Breaking changes and I have bumped the MAJOR version in `version.json`.
  - [ ] Update to documentation

## Checklist

- [x] I'm sure there are no other open Pull Requests for the same update/change
- [x] I have run `Set-AVMModule` locally to generate the supporting module files.
- [x] My corresponding pipelines / checks run clean and green without any errors or warnings

@jongio and @AlexanderSehr for notification.

### CODE REVIEW

**Comment by Menghua1** [comment]
@jongio When updating [PR#3976](https://github.com/Azure/azure-dev/pull/3976), `ptn/azd/aks` was missing some parameters when migrating, so I opened this PR to modify the parameters of `ptn/azd/aks`. Please review it, thank you.

File: avm/ptn/azd/aks/main.bicep [hunk 1]
```diff
@@ -124,6 +122,54 @@ param containerRegistryRoleName string?
...
+param systemPoolConfig object = {}
+
+@description('Optional. Custom configuration of user node pool.')
+param agentPoolConfig object = {}
+
+@description('Optional. Specifies whether the KeyvaultSecretsProvider add-on is enabled or not.')
+param enableKeyvaultSecretsProvider bool = true
+
+@description('Optional. If set to true, getting static credentials will be disabled for this cluster. This must only be used on Managed Clusters that are AAD enabled.')
+param disableLocalAccounts bool = false
```
Comment (AlexanderSehr) [inline avm/ptn/azd/aks/main.bicep:135]: WAF-wise I'd probably recommend to set this to true by default - but as it's a pattern module it's up to you. Just wanted to call it out. Feel free to resolve if you disagree

File: avm/ptn/azd/aks/main.bicep [hunk 2]
```diff
@@ -301,6 +369,10 @@ module keyVault 'br/public:avm/res/key-vault/vault:0.9.0' = {
   params: {
     name: keyVaultName
     enableTelemetry: enableTelemetry
+    enableRbacAuthorization: false
```
Comment (AlexanderSehr) [inline avm/ptn/azd/aks/main.bicep:372]: Not an option?

Comment (Menghua1) [reply avm/ptn/azd/aks/main.bicep]: Updated based on your comment

File: avm/ptn/azd/aks/main.bicep (hunk 1 above)
Comment (Menghua1) [reply avm/ptn/azd/aks/main.bicep]: Updated based on your comment.

File: avm/ptn/azd/aks/main.bicep [hunk 3]
```diff
@@ -124,6 +122,66 @@ param containerRegistryRoleName string?
 @description('Optional. The name (as GUID) of the role assignment. If not provided, a GUID will be generated.')
 param aksClusterRoleAssignmentName string?
 
+@description('Optional. Custom configuration of system node pool.')
+param systemPoolConfig object = {}
+
+@description('Optional. Custom configuration of user node pool.')
+param agentPoolConfig object = {}
```
Comment (AlexanderSehr) [inline avm/ptn/azd/aks/main.bicep:129]: You could consider adding 
`import {agentPoolType } from 'br/public:avm/res/container-service/managed-cluster:0.4.1'` in the file (note: would need to go on top of a @description block, not below) to re-use the type from the AKS module - if it satisfies your needs.

Comment (AlexanderSehr) [reply avm/ptn/azd/aks/main.bicep:129]: Would look like this:
```bicep
import {agentPoolType } from 'br/public:avm/res/container-service/managed-cluster:0.4.1'
@description('Optional. Custom configuration of system node pool.')
param systemPoolConfig agentPoolType[]?

@description('Optional. Custom configuration of user node pool.')
param agentPoolConfig agentPoolType[]?
```
IF you want to use it

Comment (Menghua1) [reply avm/ptn/azd/aks/main.bicep]: Thanks for the suggestion, updated based on your comment.

Comment (AlexanderSehr) [reply avm/ptn/azd/aks/main.bicep]: It's not exactly what I meant as you did not actually introduce a UDT which means the user needs to guess how the object must look like, but maybe something to implement at a later stage

**Comment by AlexanderSehr** [comment]
Will leave the final module review to the owner @jongio :)

File: avm/ptn/azd/aks/main.bicep (hunk 3 above)
Comment (Menghua1) [reply avm/ptn/azd/aks/main.bicep]: @AlexanderSehr Updated the code again based on your comment,

**Comment by v-xuto** [comment]
@jongio Could you hep review this PR? It blocks the progress of migration to avm.

**Review by jongio — APPROVED** [review]

**Comment by Menghua1** [comment]
@AlexanderSehr Could you please review and approve this PR?

**Comment by AlexanderSehr** [comment]
> @AlexanderSehr Could you please review this PR? If there is no issue, please approve it.

I can - but as it should be up to the module owner, let's check with @jongio whether there is a reason the PR was not merged. He should have the permissions

**Comment by v-xuto** [comment]
@jongio Could you help merge this PR? Thanks a lot.

**Comment by AlexanderSehr** [comment]
> @jongio Could you help merge this PR? Thanks a lot.

Hey @v-xuto I also reached out via teams. I'm sure we get the PR in in no time

**Pull request merged**